app/src/renderer/vuex/modules/distribution.js

```diff
@@ -0,0 +1,178 @@
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async withdrawAllRewards(
+            { rootState: { wallet }, dispatch },
+            { password, submitType }
+        ) {
+            await dispatch(`sendTx`, {
+                type: `postWithdrawDelegatorRewards`,
+                to: wallet.address,
+                password,
+                submitType
+            })
+            await dispatch(`getTotalRewards`)
+        },
+        async getRewardsFromValidator({ state, rootState: { session }, commit }, validatorAddr) {
+            state.loading = true
+            try {
+                const rewardsArray = await node.getDelegatorRewardsFromValidator(session.address, validatorAddr)
+                const rewards = coinsToObject(rewardsArray)
+                commit(`setDelegationRewards`, { validatorAddr, rewards })
+                state.error = null
+            } catch (error) {
+                commit(`notifyError`, {
+                    title: `Error getting rewards from validator`,
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async withdrawRewardsFromValidator(
+            { rootState: { wallet }, commit, dispatch },
+            { validatorAddr, password, submitType }
+        ) {
+            await dispatch(`sendTx`, {
+                type: `postWithdrawDelegatorRewardsFromValidator`,
+                to: wallet.address,
+                pathParameter: validatorAddr,
+                password,
+                submitType
+            })
+            // optimistic update
+            commit(`setDelegationRewards`, { validatorAddr, rewards: {} })
+            // update rewards
+            await dispatch(`getTotalRewards`)
+            await dispatch(`getRewardsFromValidator`, validatorAddr)
+        },
+        // TODO: move to a common parameters module
+        async getDistributionParameters({ commit }) {
+            state.loading = true
+            try {
+                const parameters = await node.getDistributionParameters()
+                commit(`setDistributionParameters`, parameters)
+                state.error = null
+            } catch (error) {
+                commit(`notifyError`, {
+                    title: `Error querying distribution parameters`,
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async getOutstandingRewards({ commit }) {
+            state.loading = true
```

Comment:
note: we are using the same variable `loading` for rewards and parameters. so they are influencing each other. this might result into inconveniences.

Reply:
ah! multiple loading properties are required then!